qiskit/transpiler/preset_passmanagers/generate_preset_pass_manager.py

```diff
@@ -226,13 +228,23 @@ def generate_preset_pass_manager(
                 raise ValueError(
                     f"Gates with 3 or more qubits ({gate}) in `basis_gates` or `backend` are "
                     "incompatible with a custom `coupling_map`. To include 3-qubit or larger "
-                    " gates in the transpilation basis, use a custom `target` instance instead."
+                    " gates in the transpilation basis, provide a custom `target` instead."
                 )
 
     if target is None:
         if backend is not None and _no_loose_constraints:
             # If a backend is specified without loose constraints, use its target directly.
             target = backend.target
+        elif _adjust_dt:
+            # If a backend is specified with loose dt, use its target and adjust the dt value.
+            target = copy.deepcopy(backend.target)
+            target.dt = dt
+            # we must update the instruction properties to internally clear the value of
+            # _instruction_durations in the target so that the new dt is saved
+            for inst_name in target.operation_names:
+                props_map = target[inst_name]
+                for qargs, props in props_map.items():
+                    target.update_instruction_properties(inst_name, qargs, props)
```

Comment:
I don't like this iteration over the target, but `target._instruction_durations` stores the value of `dt`, so the only way of truly adjusting `dt` is resetting this value (see https://github.com/Qiskit/qiskit/blob/main/qiskit/transpiler/target.py#L391-L406). This was the way I found of resetting the value of `self._instruction_durations` without directly modifying internal variables (see https://github.com/Qiskit/qiskit/blob/main/qiskit/transpiler/target.py#L364-L377). I think that for 2.1 we could add a new parameter to the `Target.duration`s method to do something like `target.durations(clear=True)` that recalculates the instruction durations if we know that `dt` has changed.

Reply:
I realized that we only need to call `update_instruction_properties` once, but I still think this can be done in a cleaner way: [d0130b0](https://github.com/Qiskit/qiskit/pull/14065/commits/d0130b07ff55a21ad2179b4ea07dd11487690cf1)